practice-two/src/components/CartItem.jsx

```diff
@@ -53,35 +56,65 @@ const ProductImage = ({ src, alt }) => (
  * @param productQuantity - The product Quantity number
  * @param removeFromCart - Function to handle remove a product to the cart
  * @param updateQuantity - Function to handle update a product to the cart
+ * @param getProductPrice - Function to get prices product
  * @returns {JSX.Element} The CartItem Component
  */
 const CartItem = ({
   product = {},
   productQuantity = {},
   updateQuantity = () => {},
   removeFromCart = () => {},
+  getProductPrice = () => {},
 }) => {
-  const { images, name, version, resolution, price } = product;
+  const { id, images, name, version, resolution, price } = product;
+
+  // Extract the first image from the images object
   const firstImage = Object.values(images)[0];
 
+  // State for managing the quantity of the product
   const [quantity, setQuantity] = useState(productQuantity);
 
+  // Convert the price from string to a numeric value
+  const parsedPrice = convertPriceToNumber(price);
+
+  // State for storing the total price of the product based on quantity
+  const [productPrice, setProductPrice] = useState(parsedPrice * quantity);
+
+  // Function to update both quantity and total price of the product
+  const updateProductPriceAndQuantity = (newQuantity) => {
+    setQuantity(newQuantity);
+
+    // Call the updateQuantity function to update the cart's quantity for this product
+    updateQuantity(id, newQuantity);
+
+    // Calculate the new total price based on the updated quantity
+    const newProductPrice = parsedPrice * newQuantity;
+    setProductPrice(newProductPrice);
+
+    // Call the getProductPrice function to update the cart's total price
+    getProductPrice(id, newProductPrice);
+  };
+
+  // Function to decrease the product quantity by 1
   const decreaseQuantity = () => {
     if (quantity > 0) {
-      const updatedQuantity = quantity - 1;
-      setQuantity(updatedQuantity);
-      updateQuantity(product.id, updatedQuantity);
+      updateProductPriceAndQuantity(quantity - 1);
     }
   };
 
+  // Function to increase the product quantity by 1
   const increaseQuantity = () => {
-    const updatedQuantity = quantity + 1;
-    setQuantity(updatedQuantity);
-    updateQuantity(product.id, updatedQuantity);
+    updateProductPriceAndQuantity(quantity + 1);
   };
 
+  // UseEffect hook to update the product's total price whenever the quantity changes
+  useEffect(() => {
```

Comment:
I don't find any reason why we need to use the useEffect, Please help check , ping me if you need help

Reply:
useEffect helps to update prices when products are added to the cart or quantity changes.
When usingEffect is not used when adding a product to the cart, the total cost will not be updated, only updated when the user increases or decreases the quantity of the product.

## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/ef012d70-60f1-4fc4-8f3e-78b2f0c704d2